src/androidTest/java/androidx/core/app/job/JobSchedulerTest.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.core.app.job
+
+import android.app.job.JobParameters
+import android.app.job.JobService
+import android.content.ComponentName
+import android.support.test.InstrumentationRegistry
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.concurrent.withLock
+
+class JobSchedulerTest {
+    @Test fun testScheduleJob() {
+        scheduleJob(context, ComponentName(context, TestService::class.java), 0) {
+            setOverrideDeadline(1)
+        }
+
+        var looperWaitCount = MAX_LOOPER_WAIT_COUNT
+        lock.withLock {
```

Comment:
All of this locking seems like it can be replaced with a simple `CountDownLatch(1)` and await with a reasonable timeout. Right now it seems overly complicated for no good reason.

Reply:
Agreed and updated. I was just following the convention I observed in the Android CTS.